src/smf/packet-smf.c

```diff
@@ -861,32 +877,36 @@ static guint get_smf_pdu_len(packet_info* inf, tvbuff_t *tvb, int offset, void *
 }
 
 /* Add a base-64 encoded string to the tree */
-static void smf_proto_add_base64_string(proto_tree *tree, int id, tvbuff_t *tvb,
+static void smf_proto_add_base64_string(proto_tree *tree, packet_info *pinfo, int id, tvbuff_t *tvb,
     int offset, int size)
 {
     char* str;
     gsize len = size;
    
     str = tvb_get_string_enc(NULL, tvb, offset, size, ENC_ASCII);
     if (len > 1) {
-        g_base64_decode_inplace(str, &len);
+        if (strlen(str) > 1) {
+            g_base64_decode_inplace(str, &len);
+        } else {
+            g_print("smf g_base64_decode_inplace failed at packet %d, strlen is < 1", pinfo->fd->num);
```

Comment:
tvn_get_string_enc returned a zero length string, is the failure here.  Not g_base64_decode_inplace which we didn't even call.

is this worthy of spaming the console with errors?  We should reserve g_print for bad errors that might indicate a bug. If you judge this to be such that's fine too.

Reply:
Without this change, we are getting the error directly from g_base64_decode_inplace() in the console without the frame number.
With a g_print() with frame number, it is an improvement.
If we do not print anything, we need to add an exception error (displayed during smf decoding) instead of a g_print().
Let me make that change...

Reply:
If we're just replacing a one g_print with a better one, that's fine.  However we should say '`g_base64_decode_inplace` failed when we didn't even call it.  We should say 'invalid base64 string found in <blah>`